assets/js/components/data.js

```diff
@@ -59,6 +59,28 @@ const lazilySetupLocalCache = () => {
 	}
 };
 
+/**
+ * Gets a copy of the given data request object with the data.dateRange populated via filter, if not set.
+ * Respects the current dateRange value, if set.
+ *
+ * @param {Object} originalRequest Data request object.
+ * @return {Object} New data request object.
+ */
+const requestWithDateRange = ( originalRequest ) => {
+	/**
+	 * Filter the date range used for queries.
+	 *
+	 * @param String The selected date range. Default 'Last 28 days'.
+	 */
+	const dateRangeSlug = stringToSlug( applyFilters( 'googlesitekit.dateRange', __( 'Last 28 days', 'google-site-kit' ) ) );
```

Comment:
This filter is now triggered for every `request` object. Can you change it so that it still triggers once like before, and the value is passed to this helper function instead?

Reply:
Updated 👍